src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs

```diff
@@ -77,11 +77,60 @@ public override void VisitField(IFieldSymbol symbol)
             }
         }
 
+        private static bool ShouldPropertyDisplayReadOnly(IPropertySymbol property)
+        {
+            if (property.ContainingType?.IsReadOnly == true)
+            {
+                return false;
+            }
+
+            // If at least one accessor is present and all present accessors are readonly, the property should be marked readonly.
+
+            var getMethod = property.GetMethod;
+            if (getMethod is object && !ShouldMethodDisplayReadOnly(getMethod, property))
+            {
+                return false;
+            }
+
+            var setMethod = property.SetMethod;
+            if (setMethod is object && !ShouldMethodDisplayReadOnly(setMethod, property))
+            {
+                return false;
+            }
+
+            return getMethod is object || setMethod is object;
+        }
+
+        private static bool ShouldMethodDisplayReadOnly(IMethodSymbol method, IPropertySymbol propertyOpt = null)
+        {
+            if (method.ContainingType?.IsReadOnly == true)
+            {
+                return false;
+            }
+
+            if (method is SourcePropertyAccessorSymbol sourceAccessor && propertyOpt is SourcePropertySymbol sourceProperty)
```

Comment:
Should this be using method.AssociatedSymbol to get the source property instead? My guess being one of three things are true:

1. ShouldMethodDisplayReadOnly gets called directly with an accessor and not just from ShouldPropertyDisplayReadOnly, at which point the callers are having to get the source property,
2. This is only being passed by ShouldPropertyDisplayReadOnly, at which point we could just inline this into that, or:
3. We're trying to do something fancier here, and we probably should have a comment.

Reply:
I believe (1) is the answer here. See the `AddAccessor` method in this file.